apps/prairielearn/python/python_helper_sympy_test.py

```diff
@@ -272,17 +273,36 @@ def test_bad_assumption_conversion(
 
 
 class TestExceptions:
-    VARIABLES = ["n"]
-
-    COMPLEX_CASES = ["i", "5 * i", "j", "I"]
-    NO_FLOATS_CASES = ["3.5", "4.2n", "3.5*n", "3.14159*n**2", "sin(2.3)"]
-    INVALID_EXPRESSION_CASES = ["5==5", "5!=5", "5>5", "5<5", "5>=5", "5<=5"]
-    INVALID_FUNCTION_CASES = ["eval(n)", "f(n)", "g(n)+cos(n)", "dir(n)", "sin(f(n))"]
-    INVALID_VARIABLE_CASES = ["x", "exp(y)", "z*n"]
-    FUNCTION_NOT_CALLED_CASES = ["2+exp", "cos*n"]
-    INVALID_PARSE_CASES = ["(", "n**", "n**2+", "!"]
-    INVALID_ESCAPE_CASES = ["\\", "n + 2 \\", "2 \\"]
-    INVALID_COMMENT_CASES = ["#", "n + 2 # comment", "# x"]
+    VARIABLES: typing.ClassVar = ["n"]
+
+    COMPLEX_CASES: typing.ClassVar = ["i", "5 * i", "j", "I"]
+    NO_FLOATS_CASES: typing.ClassVar = [
+        "3.5",
+        "4.2n",
+        "3.5*n",
+        "3.14159*n**2",
+        "sin(2.3)",
+    ]
+    INVALID_EXPRESSION_CASES: typing.ClassVar = [
+        "5==5",
+        "5!=5",
+        "5>5",
+        "5<5",
+        "5>=5",
+        "5<=5",
+    ]
+    INVALID_FUNCTION_CASES: typing.ClassVar = [
+        "eval(n)",
+        "f(n)",
+        "g(n)+cos(n)",
+        "dir(n)",
+        "sin(f(n))",
+    ]
+    INVALID_VARIABLE_CASES: typing.ClassVar = ["x", "exp(y)", "z*n"]
```

Comment:
These annotations are required on mutable fields so that you don't accidentally modify them anywhere.

Reply:
https://docs.astral.sh/ruff/rules/mutable-class-default/ lists an alternative:

> When mutability is not required, values should be immutable types, like tuple or frozenset.

Would using tuples for all of these be better?

Reply:
Resolved in [34a99e4](https://github.com/PrairieLearn/PrairieLearn/pull/11111/commits/34a99e4635016f5bad224dbbf9f71e0e7183aca4).